pkg/build/strategies/sti/sti.go

```diff
@@ -642,7 +642,7 @@ func (builder *STI) Execute(command string, user string, config *api.Config) err
 	}
 	// Do not wait for source input if the container times out.
 	// FIXME: this potentially leaks a goroutine.
-	if !util.IsTimeoutError(err) {
+	if err == nil {
```

Comment:
Hmm, can I propose the idiom of indenting error flow and keeping the happy path vertical?

```go
if err != nil {
	return err
}

wg.Wait()
return nil
```

(and adjust the comment above the code accordingly)

WDYT?